### PR TITLE
Fix GetThemeMargins p/invoke signature

### LINKED ISSUE #526: Invalid PInvoke definition for SafeNativeMethods.GetThemeMargins()
`VisualStyleRenderer.GetMargins(...)` throws an AccessViolation because of an invalid PInvoke definition for `SafeNativeMethods.GetThemeMargins(...)`
According to https://docs.microsoft.com/en-us/windows/desktop/api/uxtheme/nf-uxtheme-getthememargins the LPCRECT parameter is missing. 
The method is currently not used as far as I can see, but its wrong behavior is mentioned in some code comments.

### PR DESCRIPTION
**Pull request by nikeee**
According to the MS docs, the parameter `LPCRECT prc` is missing:
```c
THEMEAPI GetThemeMargins(
  HTHEME  hTheme,
  HDC     hdc,
  int     iPartId,
  int     iStateId,
  int     iPropId,
  LPCRECT prc,
  MARGINS *pMargins
);
```

This PR adds this missing parameter.

Fixes #526

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/VisualStyles/VisualStyleRenderer.cs [hunk 1]
```diff
@@ -822,7 +822,7 @@ public Font GetFont(IDeviceContext dc, FontProperty prop)
 
             using( WindowsGraphicsWrapper wgr = new WindowsGraphicsWrapper( dc, AllGraphicsProperties ) ) {
                 HandleRef hdc = new HandleRef( wgr, wgr.WindowsGraphics.DeviceContext.Hdc );
-                lastHResult = SafeNativeMethods.GetThemeMargins( new HandleRef( this, Handle ), hdc, part, state, (int) prop, ref margins );
+                lastHResult = SafeNativeMethods.GetThemeMargins( new HandleRef( this, Handle ), hdc, part, state, (int) prop, default(NativeMethods.COMRECT), ref margins );
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/VisualStyles/VisualStyleRenderer.cs]: Could you please add a simple unit test that invokes VisualStyleRenderer.GetMargins?

Comment (nikeee) [reply src/System.Windows.Forms/src/System/Windows/Forms/VisualStyles/VisualStyleRenderer.cs]: I added a test in 1d2e3fa. Does it suffice?

**Comment by nikeee** [comment]
I added a test in 1d2e3fa. Does it suffice?

**Comment by sharwell** [comment]
@nikeee Did it throw an exception prior to this?

File: src/System.Windows.Forms/tests/UnitTests/VisualStyleRendererTests.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+using System.Windows.Forms.VisualStyles;
+
+namespace System.Windows.Forms.Tests
+{
+    public class VisualStyleRendererTests
+    {
+        [Fact]
```
Comment (sharwell) [inline src/System.Windows.Forms/tests/UnitTests/VisualStyleRendererTests.cs]: 💡 I'm not sure if `WorkItemAttribute` is defined for this project already, but it's very convenient for documenting tests created in response to specific bugs.

```suggestion
        [Fact]
        [WorkItem(526, "https://github.com/dotnet/winforms/issues/526")]
```

File: src/System.Windows.Forms/tests/UnitTests/VisualStyleRendererTests.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        public void VisualStyleRenderer_GetMargins()
+        {
+            var renderer = new VisualStyleRenderer(VisualStyleElement.Button.PushButton.Normal);
+
+            using (var form = new System.Windows.Forms.Form())
+            using (var graphics = form.CreateGraphics())
+            {
+                // GetMargins should not throw an exception.
+                // See Issue #526 on GitHub.
```
Comment (sharwell) [inline src/System.Windows.Forms/tests/UnitTests/VisualStyleRendererTests.cs]: 💡 It helps to either use `dotnet/winforms#526` or a full link in cases like this.

```suggestion
                // See Issue https://github.com/dotnet/winforms/issues/526 on GitHub.
```

File: src/System.Windows.Forms/src/System/Windows/Forms/VisualStyles/VisualStyleRenderer.cs (hunk 1 above)
Comment (sharwell) [inline src/System.Windows.Forms/src/System/Windows/Forms/VisualStyles/VisualStyleRenderer.cs]: 💡 `default(NativeMethods.COMRECT)` (and the C# 7.1 equivalent `default`) suggests to the reader that `COMRECT` is a struct type. However, this is not correct. The preferred form would be the following:

```suggestion
                lastHResult = SafeNativeMethods.GetThemeMargins( new HandleRef( this, Handle ), hdc, part, state, (int) prop, prc: null, ref margins );
```

File: src/System.Windows.Forms/tests/UnitTests/VisualStyleRendererTests.cs (hunk 2 above)
Comment (nikeee) [reply src/System.Windows.Forms/tests/UnitTests/VisualStyleRendererTests.cs]: The project does not contain any uses of `WorkItemAttribute`, so I think it is not (yet) defined in this project.

File: src/System.Windows.Forms/tests/UnitTests/VisualStyleRendererTests.cs (hunk 3 above)
Comment (nikeee) [reply src/System.Windows.Forms/tests/UnitTests/VisualStyleRendererTests.cs]: Done!

**Comment by nikeee** [comment]
Thanks for you review!

> Did it throw an exception prior to this?

Yes, it threw an `AccessViolationException` due to the wrong p/invoke signature.

> I would suggest removing Fix #526 from the title and instead adding Fixes #526 as the last line of the pull request description before merging this.

Done.

The CI release build failed after applying the last commit (debug build worked). I don't know why, though.

**Comment by zsd4yr** [comment]
LGTM

**Comment by zsd4yr** [comment]
Thanks @nikeee

**Pull request merged**